components/EditStoryForm/index.js

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from "react";
+import Image from "next/image";
+import {
+  Form,
+  Input,
+  TextArea,
+  Button,
+  Label,
+} from "../NewStoryForm/StyledNewStoryForm";
+//import { useImmerLocalStorageState } from "@/lib/hook/useImmerLocalStorageState";
+//import produce from "use-immer";
+export default function EditStoryForm({ story, onUpdate }) {
+  const [title, setTitle] = useState(story.title);
+  const [coverImage, setCoverImage] = useState(null);
+  const [textContent, setTextContent] = useState(story.textContent);
+  const [isUploading, setIsUploading] = useState(false);
+
+  function handleTitleChange(event) {
+    setTitle(event.target.value);
+  }
+
+  function handleImageChange(event) {
+    const file = event.target.files[0];
+    setCoverImage(file);
+  }
+
+  function handleTextChange(event) {
+    setTextContent(event.target.value);
+  }
+
+  async function handleSubmitEdit(event) {
+    event.preventDefault();
+    setIsUploading(true);
+
+    const formData = new FormData();
+    if (coverImage) {
+      formData.append("file", coverImage);
+      formData.append("upload_preset", process.env.NEXT_PUBLIC_UPLOAD_PRESET);
+    }
+    try {
+      let coverImageUrl = story.coverImage;
+      if (coverImage) {
+        const response = await fetch(
+          `https://api.cloudinary.com/v1_1/${process.env.NEXT_PUBLIC_CLOUDNAME}/upload`,
+          {
+            method: "POST",
+            body: formData,
+          }
+        );
+        const json = await response.json();
+        coverImageUrl = json.secure_url;
+      }
+
+      //   const updatedStory = produce(story, (draft) => {
+      //     draft.title = title;
+      //     draft.coverImage = coverImageUrl;
+      //     draft.textContent = textContent;
+      //     draft.dateModified = new Date().toLocaleDateString();
+      //   });
+
+      const updatedStory = {
+        ...story,
+        title,
+        coverImage: coverImageUrl,
+        textContent,
+        dateModified: new Date().toLocaleDateString(),
```

Comment:
Better store dates as timestamps (i.e. `new Date().toISOString()`), so one can render them more flexibly in the future ^^